pkg/odo/cli/deploy/deploy.go

```diff
@@ -65,7 +65,7 @@ func (o *DeployOptions) Complete(cmdline cmdline.Cmdline, args []string) (err er
 		return err
 	}
 	if isEmptyDir {
-		return errors.New("this command cannot run in an empty directory, you need to run it in a directory containing source code")
+		return errors.New("this command cannot run in an empty directory, run the command in a directory containing source code or initialize using 'odo init'")
```

Comment:
Can't we make define a sentinel error for this one in the component pkg? Something like the following - 
```go
var EmptyDirError := errors.New("this command cannot run in an empty directory, run the command in a directory containing source code or initialize using 'odo init'")
```

Reply:
Hey @valaparthvi I want to try and keep the error logging consistent across `odo` so I prefer not creating a sentinal error despite it being copy and pasted in two places...

Preferably because using those types of errors are unconventional and I want to make this PR more of a UI/UX change rather than adding to pkg/component instead of pkg/cli

Reply:
In that case,
/lgtm
/approve